src/components/ContextBreadcrumbs.tsx

```diff
@@ -43,16 +83,43 @@ const ContextBreadcrumbs = ({ homeContext, simplePath }: ContextBreadcrumbProps
       - The "c/d" context will render "d" as a thought and "c" as the breadcrumbs.
     */
       ? !homeContext ? <HomeLink color='gray' size={16} style={{ position: 'relative', left: -5, top: 2 }} /> : null
-      : simplePath.map((thoughtRanked, i) => {
-        const subthoughts = ancestors(simplePath, thoughtRanked) as SimplePath
-        return <React.Fragment key={i}>
-          <Link simplePath={subthoughts} />
-          <Superscript simplePath={subthoughts} />
-          {i < simplePath.length - 1 ? <span className='breadcrumb-divider'> • </span> : null}
-        </React.Fragment>
-      })
+      : <TransitionGroup>
+        {overflowArray.map((thoughtRanked, i) => {
+          const subthoughts = ancestors(simplePath, thoughtRanked) as SimplePath
+          return <CSSTransition key={i} timeout={600} classNames='fade-600'>
+
+            {!thoughtRanked.isOverflow ?
+              <span>
+                {i > 0 ? <span className='breadcrumb-divider'> • </span> : null}
+                {subthoughts && <Link simplePath={subthoughts} label={thoughtRanked.label} />}
+                {subthoughts && <Superscript simplePath={subthoughts} />}
+              </span>
+              :
+              <span>
+                <span className='breadcrumb-divider'> • </span>
+                <span onClick={() => setEllipsize(false)} style={{ cursor: 'pointer' }}> ... </span>
+              </span>
+            }
+          </CSSTransition>
+        })}
+      </TransitionGroup>
     }
   </div>
 }
 
+/**
+ * Varaint of ContextBreadcrumbs for recently edited with collapsing overflow.
+ */
+export const RecentlyEditedBreadcrumbs = connect(mapStateToProps)((props: ContextBreadcrumbProps & ReturnType<typeof mapStateToProps>) => {
+
+  const parentSimplePath = props.simplePath.slice(0, -1) as SimplePath
+  const simplePath = props.simplePath
+
+  return <div className='recently-edited-breadcrumbs'>
+    <ContextBreadcrumbs {...props} simplePath={parentSimplePath}/>
+    <Link simplePath={simplePath} label={headValue(simplePath)} />
+    <Superscript simplePath={simplePath} />
+  </div>
+})
```

Comment:
Let's stick to our convention of one exported component per file.

Reply:
Since `RecentlyEditedBreadcrumbs` uses `mapStateToProps` to calculate `simplePath`, `ContextBreadcrumbs` doesn't need to calculate simplified path again. Defining `RecentlyEditedBreadcrumbs` in this file, we can use `ContextBreadcrumbs` that is not connected to redux.

I will create a separate file for `RecentlyEditedBreadcrumbs`. However same `mapStateToProps` will be exeuted twice. Once for both connected components.

Reply:
That's a fair point. Another way to achieve that is to export the unconnected `ContextBreadcrumbs` so that it can be imported "laterally" by `RecentlyEditedBreadcrumbs`. That allows you to avoid the extra `mapStateToProps` while allowing `RecentlyEditedBreadcrumbs` to be a default import.